pkg/remote/terraform/lockfile.go

```diff
@@ -0,0 +1,223 @@
+package terraform
```

Comment:
It could be nice to put this in `pkg/terraform/lock` instead of remote, this is not really related to driftctl remote stuff.

We should probably only expose a single method from this pkg to return the related provider config from this pkg.

We may return an error (or terraform diags) too and log it in debug